taipy/gui/_renderers/json.py

```diff
@@ -20,35 +22,60 @@
 from .._warnings import _warn
 from ..icon import Icon
 from ..utils import _date_to_string, _MapDict, _TaipyBase
+from ..utils.singleton import _Singleton
 
 
-def _default(o):
-    if isinstance(o, Icon):
-        return o._to_dict()
-    if isinstance(o, _MapDict):
-        return o._dict
-    if isinstance(o, _TaipyBase):
-        return o.get()
-    if isinstance(o, (datetime, date, time)):
-        return _date_to_string(o)
-    if isinstance(o, Path):
-        return str(o)
-    if isinstance(o, FunctionType):
-        return o.__name__
-    if isinstance(o, LambdaType):
-        return str(o)
-    try:
-        raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
-    except Exception as e:
-        _warn("Exception in JSONEncoder", e)
+class JsonResolver(ABC):
+    def register(self):
+        _TaipyJsonResolver().register(self)
+
+    @abstractmethod
+    def resolve(self, o) -> t.Union[t.Any, None]:
         return None
 
 
+class _DefaultJsonResolver(JsonResolver):
+    def resolve(self, o):
+        if isinstance(o, Icon):
+            return o._to_dict()
+        if isinstance(o, _MapDict):
+            return o._dict
+        if isinstance(o, _TaipyBase):
+            return o.get()
+        if isinstance(o, (datetime, date, time)):
+            return _date_to_string(o)
+        if isinstance(o, Path):
+            return str(o)
+        if isinstance(o, FunctionType):
```

Comment:
Are the 2 (FunctionType and lambdaType) new ?
Could this have an impact on taipy-gui : like no error when you get a function when your not supposed to ?

Reply:
It was new. I added it on a recent PR for the frontend decopule. This is just json encoding so I don't think it will affect anything. Previously you will get an error and the result is None. Now you get the function name with no error which is fine tbh

Reply:
less error reporting but I guess it's OK
Or should those be handled by the chalk'it Json resolver ?

Reply:
I like that proposal!. For normal taipy app you will never encounter this issue. Only for custom frontend will you encounter this when getting the function list. 
Let's wait for what @FabienLelaquais says.